libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1272,23 +1272,23 @@ string YulUtilFunctions::storageByteArrayPopFunction(ArrayType const& _type)
 	});
 }
 
-string YulUtilFunctions::storageArrayPushFunction(ArrayType const& _type)
+string YulUtilFunctions::storageArrayPushFunction(ArrayType const& _type, TypePointer _fromType)
 {
 	solAssert(_type.location() == DataLocation::Storage, "");
 	solAssert(_type.isDynamicallySized(), "");
 
```

Comment:
I think this needs an assertion that if `_fromType` is a value type, then it is the same as `_type.baseType()`. Another solution would be to add a conversion function.

Reply:
I would rather add assertion that types are the same for now (in case that `_fromType` is value type)